src/fields.js

```diff
@@ -1450,11 +1465,11 @@ x3dom.fields.SFMatrix4f.prototype.spectralDecompose = function ( SO, k )
 
                 for ( var j = 2; j >= 0; --j )
                 {
-                    var a = SO[ "_" + j + p ];
-                    var b = SO[ "_" + j + q ];
+                    var a = SO.at( j, p );
+                    var b = SO.at( j, q );
 
-                    SO[ "_" + j + p ] -= s * ( b + tau * a );
-                    SO[ "_" + j + q ] += s * ( a - tau * b );
+                    SO.setAt( j, p, SO.at( j, p ) - s * ( b + tau * a ) );
+                    SO.setAt( j, q, SO.at( j, q ) + s * ( a - tau * b ) );
                 }
             }
         }
```

Comment:
Ok. This could be perhaps a bit slower but hopefully not noticeable. These functions are used for default navigation, eg. every frame. Please test just a view example scenes with these changes.

Reply:
Did  you have a chance to these changes ?

Reply:
oh well, let's just go for it then.